features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NRF5x/source/btle/btle.cpp

```diff
@@ -369,6 +369,53 @@ static void btle_handler(ble_evt_t *p_ble_evt)
             break;
         }
 
+
+#if  (NRF_SD_BLE_API_VERSION >= 5)
+#ifndef S140
+        // Handle PHY upgrade request
+        case BLE_GAP_EVT_PHY_UPDATE_REQUEST:
+        {
+            ble_gap_phys_t const phys =
+            {
+                /* rx_phys */ BLE_GAP_PHY_AUTO,
+                /* tx_phys */ BLE_GAP_PHY_AUTO,
+            };
+            ASSERT_STATUS_RET_VOID( sd_ble_gap_phy_update(p_ble_evt->evt.gap_evt.conn_handle, &phys) );
+            break;
+        }
+#endif
+
+        	// Handle Data length negotiation request
+        case BLE_GAP_EVT_DATA_LENGTH_UPDATE_REQUEST:
+        {
+            ble_gap_evt_t       const * p_gap_evt = &p_ble_evt->evt.gap_evt;
+            uint8_t const data_length_peer =
+                p_gap_evt->params.data_length_update_request.peer_params.max_tx_octets;
+
+            const uint8_t max_data_length = NRF_SDH_BLE_GATT_MAX_MTU_SIZE + 4 /* L2CAP header size */;
```

Comment:
The L2CAP header size is defined with `L2CAP_HDR_LEN`.

Reply:
Actually this is defined in `nrf_ble_gatt.c` in the Nordic SDK so not available in that scope, that's why I put it like that.